netsuite/11.3/modules/ROOT/pages/netsuite-examples.adoc

```diff
@@ -0,0 +1,790 @@
+		<netsuite:token-based-authentication-connection consumerKey="${netsuite.consumerKey}" consumerSecret="${netsuite.consumerSecret}" tokenId="${netsuite.tokenId}" tokenSecret="${netsuite.tokenSecret}" account="${netsuite.account}"/>
+	</netsuite:config>
+
+    <flow name="customer-advanced-search-flow">
+		<http:listener doc:name="Listener" config-ref="HTTP_Listener_config" path="customer" />
+		<ee:transform doc:name="Search Criteria">
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/xml
+ns ns0 urn:messages_2020_2.platform.webservices.netsuite.com
+ns ns01 urn:common_2020_2.platform.webservices.netsuite.com
+ns ns02 urn:core_2020_2.platform.webservices.netsuite.com
+ns ns03 urn:relationships_2020_2.lists.webservices.netsuite.com
+ns xsi http://www.w3.org/2001/XMLSchema-instance
+---
+{
+	ns0#search: {
+		ns0#searchRecord @("xmlns:ns03": ns03, xsi#"type": "ns03:CustomerSearchAdvanced"): {
+			ns03#criteria: {
+				ns03#basic: {
+					ns01#stage @(operator: "anyOf"): {
+						ns02#searchValue: [attributes.queryParams["stage"]]
+					},
+					ns01#dateCreated @(operator: "after"): {
+						ns02#searchValue: "2015-01-01T00:00:00.000-08:00"
+					}
+				}
+			}
+		}
+	}
+}]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+		<netsuite:search doc:name="Search Customer Advanced" config-ref="tokenConfig" returnSearchColumns="false" key="CustomerSearchAdvanced"/>
+		<foreach doc:name="For Each">
+			<ee:transform doc:name="Transform to JSON">
+			<ee:message>
+				<ee:set-payload><![CDATA[%dw 2.0
+output application/json
+ns ns0 urn:core_2020_2.platform.webservices.netsuite.com
+---
+payload.ns0#record
+]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+			<logger level="INFO" doc:name="Logger" message="#[payload]" />
+		</foreach>
+	</flow>
+</mule>
+----
+
+== Custom Records and Fields
+
+The following example shows how to add, get, and delete custom records with
+custom fields using DataSense.
+
+The following screenshot shows the Studio app flow for this example:
+
+image::custom-record-flow.png[App flow for the Custom Records example]
+
+Creating this example involves adding the *Add* operation, *Set Variable*
+component, *Get* operation, and *Delete* operation.
+
+[[add-custom-record]]
+=== Add a Custom Record
+
+Add a custom record with custom fields using the *Add*
+operation.
+
+Include the following attributes to add a custom record:
+
+* *separators*
++
+Specifies a separator character. The separator in the following example is `__`.
+
+* *recordType*
++
+References the custom record type. The recordType in the following example is `customRecordType`.
+
+* *scriptId*
++
+References the script ID of the record. The scriptId in the following example is `customrecord_2663_bank_details`.
+
+* *internalId*
++
+References the internal ID of the record. The internalId in the following example is `374`.
+
+* *XSI type*
++
+Specify the XSI type as *CustomRecordRef* to add a custom record.
```

Comment:
It looks like this is set to *CustomRecord* in the script below

Reply:
Looks like the script has the type as "ns01:CustomRecordRef"